migration-engine/migration-engine-tests/tests/migrations/indexes.rs

```diff
@@ -1035,3 +1035,100 @@ fn do_not_overwrite_fulltext_index_without_preview_feature(api: TestApi) {
 
     api.schema_push_w_datasource(dm).send().assert_no_steps();
 }
+
+#[test_connector(tags(Mysql), preview_features("fullTextIndex"))]
+fn adding_fulltext_index_to_an_existing_column(api: TestApi) {
+    let dm = formatdoc! {r#"
+        {}
+
+        generator client {{
+          provider = "prisma-client-js"
+          previewFeatures = ["fullTextIndex"]
+        }}
+
+        model A {{
+          id Int    @id
+          a  String @db.Text
+          b  String @db.Text
+        }}
+    "#, api.datasource_block()};
```

Comment:
I think you could use api.datamodel_with_provider which pulls in the preview flags automatically.